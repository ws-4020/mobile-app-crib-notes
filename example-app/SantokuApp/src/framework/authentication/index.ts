export {useAuthenticationService, ActiveAccountIdNotFoundError, PasswordNotFoundError} from './AuthenticationService';
