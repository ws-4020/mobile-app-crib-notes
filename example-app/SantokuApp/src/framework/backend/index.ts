export * from './WithReactQuery';
export * from './useDefaultOptions';
export {refreshCsrfToken} from './useCustomInstance';
