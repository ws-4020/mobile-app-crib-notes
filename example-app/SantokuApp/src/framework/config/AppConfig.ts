import {Config} from 'react-native-config';

/**
 * 環境設定値を取得できます。
 *
 * ```ts
 * import {config} from 'framework/config';
 *
 * console.log(Config.appNameHome);
 * ```
 */
export abstract class AppConfig {
  static get appIdFlavorSuffix(): string {
    return Config.APP_ID_FLAVOR_SUFFIX;
  }

  static get appNameHome(): string {
    return Config.APP_NAME_HOME;
  }

  static get provisioningProfileFlavor(): string {
    return Config.PROVISIONING_PROFILE_FLAVOR;
  }
}
