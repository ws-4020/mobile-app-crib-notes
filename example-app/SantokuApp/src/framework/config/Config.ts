import {Config as RNConfig} from 'react-native-config';

/**
 * 環境設定値を取得できます。
 *
 * ```ts
 * import {config} from 'framework/config';
 *
 * console.log(Config.appNameHome);
 * ```
 */
class Config {
  /**
   * `APP_ID_FLAVOR_SUFFIX`の設定値を返します。
   */
  get appIdFlavorSuffix(): string {
    return RNConfig.APP_ID_FLAVOR_SUFFIX;
  }

  /**
   * `APP_NAME_HOME`の設定値を返します。
   */
  get appNameHome(): string {
    return RNConfig.APP_NAME_HOME;
  }

  /**
   * `PROVISIONING_PROFILE_FLAVOR`の設定値を返します。
   */
  get provisioningProfileFlavor(): string {
    return RNConfig.PROVISIONING_PROFILE_FLAVOR;
  }
}
export const config = new Config();
