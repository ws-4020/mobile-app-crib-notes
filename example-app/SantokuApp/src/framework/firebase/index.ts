import {firebaseConfig} from './FirebaseConfig';

export {firebaseConfig};
