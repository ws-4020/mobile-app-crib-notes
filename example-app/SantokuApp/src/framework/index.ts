export {AuthnService, ActiveAccountIdNotFoundError, PasswordNotFoundError} from './authentication';
export {api, csrfToken} from './backend';
export {AppConfig} from './config';
export {firebaseConfig, FirebaseConfig} from './firebase';
export type {Logger, LoggerOptions} from './logging';
export {log, createLogger} from './logging';
export {m, loadMessages, BundledMessagesLoader} from './message';
export {launchedId, generatePassword, useIsMounted} from './utilities';
