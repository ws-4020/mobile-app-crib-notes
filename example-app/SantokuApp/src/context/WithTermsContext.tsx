import {AppInitialData} from 'framework/initialize/types';
import {useIsMounted} from 'framework/utilities';
import React, {useCallback, useState} from 'react';

import {SetTermsAction, SetTermsContextProvider} from './useSetTermsContext';
import {TermsContext, TermsContextProvider} from './useTermsContext';

const initialTermsContext = {};

function WithTermsContext(props: {initialData: AppInitialData; children: React.ReactNode}) {
  const isMounted = useIsMounted();
  const [state, setState] = useState<TermsContext>(props.initialData.accountData.terms ?? initialTermsContext);

  const setStartup = useCallback(
    (setStateAction: SetTermsAction) => {
      if (isMounted()) {
        setState(setStateAction);
      }
    },
    [isMounted],
  );

  return (
    <TermsContextProvider value={state}>
      <SetTermsContextProvider value={setStartup}>{props.children}</SetTermsContextProvider>
    </TermsContextProvider>
  );
}

export {WithTermsContext};
