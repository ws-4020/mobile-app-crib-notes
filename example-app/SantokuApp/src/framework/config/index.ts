import {config} from './Config';

export {config};
