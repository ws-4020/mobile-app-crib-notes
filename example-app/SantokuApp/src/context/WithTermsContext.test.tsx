import {render} from '@testing-library/react-native';
import {AppConfig} from 'framework/config';
import React from 'react';
import {Text} from 'react-native';

import {WithTermsContext} from './WithTermsContext';

describe('WithTermsContext', () => {
  const ChildComponent: React.FC = () => {
    return <Text testID="test">test</Text>;
  };

  it('WithTermsContextを子要素を含めて正常にrenderできること', () => {
    const initialData = {
      accountData: {
        terms: {
          termsOfServiceAgreementStatus: {hasAgreedValidTermsOfService: true, agreedTermsOfServiceVersion: '1.0.0'},
          termsOfService: {latestTermsOfServiceVersion: '1.0.0', url: AppConfig.termsUrl},
        },
      },
    };
    const withTermsContext = render(
      <WithTermsContext initialData={initialData}>
        <ChildComponent />
      </WithTermsContext>,
    );

    expect(withTermsContext.queryByTestId('test')).not.toBeNull();
    expect(withTermsContext).toMatchSnapshot();
  });
});
