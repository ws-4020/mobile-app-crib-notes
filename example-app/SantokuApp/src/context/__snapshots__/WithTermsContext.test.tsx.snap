// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithTermsContext WithTermsContextを子要素を含めて正常にrenderできること 1`] = `
<Text
  testID="test"
>
  test
</Text>
`;
